### ISSUE TITLE
windows_share returning WIN32OLERuntimeError

### ISSUE THREAD

**Issue by ooOOJavaOOoo** [chef-cookbooks/windows#482]
### Cookbook version
windows 3.1.1

### Chef-client version
12.19.36

### Platform Details
Windows Server 2012R2 local kitchen instance

### Scenario:
Creating a file share in windows using the windows_share resource

### Steps to Reproduce:
metadata.rb
```depends 'windows'```
update default.rb
```
directory 'd:\siebelfs' do
  action :create
end
windows_share 'd:\siebelfs' do
  action :create
  path 'd:\siebelfs'
  full_users [user_credential[:user]]
end
```
### Expected Result:
Expected directory be created with the share enabled

### Actual Result:
           ================================================================================
           Error executing action `create` on resource 'windows_share[d:\siebelfs]'
           ================================================================================

           WIN32OLERuntimeError
           --------------------
           (in OLE method `Count': )
        OLE error code:80041017 in SWbemObjectSet
          Invalid query
        HRESULT error code:0x80020009
          Exception occurred.

**Comment by trickyearlobe** [comment]
@ooOOJavaOOoo you can't put the drive letter in the share name.

Try this instead

```
windows_share 'siebelfs' do
  action :create
  path 'd:\siebelfs'
  full_users [user_credential[:user]]
end
```

**Comment by ooOOJavaOOoo** [comment]
================================================================================
           Error executing action `create` on resource
'windows_share[siebelfs]'

 ================================================================================

           Chef::Exceptions::ValidationFailed
           ----------------------------------
           path is required

           Resource Declaration:
           ---------------------
           # In
C:/Users/ADMINI~1/AppData/Local/Temp/kitchen/cache/cookbooks/role_solar_appserver/recipes/default.rb

            47: windows_share 'siebelfs' do
            48:   action :create
            49:   path 'd:\siebelfs'
            50:   full_users ['Everyone']
            51: end
            52:

**Comment by trickyearlobe** [comment]
@ooOOJavaOOoo 
Hmmm... different error now... possibly a bug in chef-client ??

In the custom resource `resources/share.rb:24` path is a property which I believe is correctly defined as `property :path, String, required: true` but it seems to be ignoring the fact you provided it. I can reproduce it on my local machine.

I can see there was a commit `fc2691f` in `3.1.1` which changed that property to `required` instead of raising an exception that it was missing. If I use the `3.1.0` version of the cookbook it converges successfully but I don't get a share created.

**Comment by mdpdesign** [comment]
Hi guys, 

I'm also having the issue with the **windows_share** resource. I tested few chef-client versions in Test Kitchen: `13.2.20`, `13.1.31`, `12.21.3`, `12.0.0` on `Windows 7`, `Windows 10`, `Windows 2012R2` with no luck.

No matter if I use paths like: `C:\Candidates` or `C:/Candidates` I always get error that the `path` parameter is required. I use `windows` cookbook version `3.1.1`

```================================================================================                                                            
Error executing action `create` on resource 'windows_share[Candidates]'                                                                     
================================================================================                                                            
                                                                                                                                            
Chef::Exceptions::ValidationFailed                                                                                                          
----------------------------------                                                                                                          
path is required                                                                                                                            
                                                                                                                                            
Resource Declaration:                                                                                                                       
---------------------                                                                                                                       
# In C:/Users/vagrant/AppData/Local/Temp/kitchen/cache/cookbooks/s5k_build_machine/recipes/shared_folders.rb                                
                                                                                                                                            
 10: windows_share "Candidates" do                                                                                                          
 11:   action :create                                                                                                                       
 12:   path 'C:/Candidates'                                                                                                                 
 13:   full_users ['Administrator']                                                                                                         
 14:   read_users ['Users']                                                                                                                 
 15: end                                                                                                                                    
 16:                                                                                                                                        
                                                                                                                                            
Compiled Resource:                                                                                                                          
------------------                                                                                                                          
# Declared in C:/Users/vagrant/AppData/Local/Temp/kitchen/cache/cookbooks/s5k_build_machine/recipes/shared_folders.rb:10:in `from_file'     
                                                                                                                                            
windows_share("Candidates") do                                                                                                              
  action [:create]                                                                                                                          
  retries 0                                                                                                                                 
  retry_delay 2                                                                                                                             
  default_guard_interpreter :default                                                                                                        
  declared_type :windows_share                                                                                                              
  cookbook_name "s5k_build_machine"                                                                                                         
  recipe_name "shared_folders"                                                                                                              
  path "C:/Candidates"                                                                                                                      
  full_users ["Administrator"]                                                                                                              
  read_users ["Users"]                                                                                                                      
  share_name "Candidates"                                                                                                                   
  description ""                                                                                                                            
  change_users []                                                                                                                           
end```

**Comment by challenger2010** [comment]
Duplicated this as well.  Reverting to windows 3.1.0 has fixed it for me as well.

**Comment by senecav** [comment]
Have duplicated this issue as well. As above reverting to version 3.1.0 worked for me as well.

Running versions - ChefDK - 1.4.3, kitchen - 1.16.0. chef-client - 12.19.36 & 13.1.31

**Comment by tas50** [comment]
This should be fixed in 3.1.2
